src/controller/python/chip/ble/LinuxImpl.cpp

```diff
@@ -130,7 +130,11 @@ extern "C" void * pychip_ble_start_scanning(PyObject * context, void * adapter,
         return nullptr;
     }
 
-    if (scanner->StartScan(chip::System::Clock::Milliseconds32(timeoutMs)) != CHIP_NO_ERROR)
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::DeviceLayer::PlatformMgr().LockChipStack();
+    err = scanner->StartScan(chip::System::Clock::Milliseconds32(timeoutMs));
```

Comment:
If the invariant is that StartScan should always be called with the lock held, we should add an assert to that effect in StartScan.  At the top, before any of the early returns that might make it not reach the asserts in StartTimer.  That will also be somewhat self-documenting regarding the invariant.